resolwe_bio/tools/basespace_download.py

```diff
@@ -0,0 +1,112 @@
+    parser.add_argument('--access-token-secret-path',
+                        dest='access_token_secret_path',
+                        required=True,
+                        help='BaseSpace access token secret path.')
+    args = parser.parse_args()
+
+    file_ids = args.file_ids
+    access_token = get_token_from_secret_file(args.access_token_secret_path)
+    headers = {'x-access-token': access_token}
+
+    for file_id in file_ids:
+        file_name = get_file_name(session, file_id, headers)
+        download_file(session, file_id, file_name, headers)
+        print('filename={}'.format(file_name))
+
+
+def on_exit(session):
+    """Clean up function called on exit."""
+    session.close()
+
+
+def get_token_from_secret_file(secret_file_path):
+    """Read secret file to obtain access token."""
+    try:
+        with open(secret_file_path, 'r') as f:
+            return f.readline()
+    except FileNotFoundError:
+        raise BaseSpaceDownloadError('Secret file not found')
+    except PermissionError:
+        raise BaseSpaceDownloadError('No permissions to read secret file')
+
+
+def make_get_request(session, url, headers, stream=False):
+    """Make a get request."""
+    response = session.get(url, headers=headers, stream=stream)
+
+    if response.status_code == 401:
+        raise BaseSpaceDownloadError('Authentication failed on URL {}'.format(url))
+
+    if response.status_code == 404:
+        raise BaseSpaceDownloadError('BaseSpace file {} not found'.format(url))
+
+    return response
+
+
+def get_basespace_api_url():
+    """Get base BaseSpace API URL."""
+    return 'https://api.basespace.illumina.com/v1pre3'
+
+
+def get_basespace_api_file_url(file_id):
+    """Get BaseSpace API file URL."""
+    return get_basespace_api_url() + '/files/' + file_id
+
+
+def get_basespace_api_file_content_url(file_id):
+    """Get BaseSpace API file contents URL."""
+    return get_basespace_api_file_url(file_id) + '/content'
+
+
+def get_file_name(session, file_id, request_headers):
+    """Get file name."""
+    response = make_get_request(session, get_basespace_api_file_url(file_id), request_headers)
+    return response.json()['Response']['Name']
+
+
+def download_file(session, file_id, file_name, request_headers):
+    """Download BaseSpace file."""
+    response = make_get_request(session, get_basespace_api_file_content_url(file_id), request_headers, True)
+
+    try:
+        with open(file_name, 'wb') as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                f.write(chunk)
+    except FileNotFoundError:
+        raise BaseSpaceDownloadError('Could not save file to {}, due to directory not being found'.format(file_name))
+    except PermissionError:
+        raise BaseSpaceDownloadError('Could not save file to {}, due to insufficient permissions'.format(file_name))
+
+
+if __name__ == "__main__":
+    try:
+        main()
+    except Exception as e:
+        print(e)
```

Comment:
Since this tool is used by processes requiring `secrets`, this line could be problematic.

Reply:
Perhaps add an argument `--verbose` and only output the full backtrace (not just `print(e)`) if this argument is passed and only output `An error occurred while processing the Basespace download request. Use --verbose to see details.`.

Reply:
> only output the full backtrace

Do you mean not catching the exception or some other way?

Reply:
No, use the `traceback` module to print the exception.